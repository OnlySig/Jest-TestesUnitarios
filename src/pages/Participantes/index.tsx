import styled from 'styled-components';
import Banner from '../../components/Banner'
import Formulario from '../../components/Formulario';
import Rodape from '../../components/Rodape';
import ListaParticipantes from '../../components/ListaParticipantes';

const FormContainer = styled.section `
    display: flex;
    align-items: center;
    flex-direction: column;
    border: 2px solid black;
    padding: 87px 0;
    background-color: #FFF9EB;
    border-radius: 64px 64px 0 0;
    height: 50vw;
`
const FormContent = styled.div `
    position: relative;
    height: 500px;
    min-width: 730px;
`
const ListaContainer = styled.ul `
    margin-top: 30px;
    list-style: none;
    text-align: center;
    overflow-y: scroll;
    max-height: 100px;
`
const Participantes = () => {
    
    return( 
        <>
            <Banner/>
            <FormContainer>
                <FormContent>
                    <Formulario/>
                    <ListaContainer>
                        <ListaParticipantes/>
                    </ListaContainer>
                    <Rodape/>
                </FormContent>
            </FormContainer>
        </>
    )
}

export default Participantes