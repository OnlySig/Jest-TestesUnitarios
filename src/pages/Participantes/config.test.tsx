import { render } from "@testing-library/react"
import { RecoilRoot } from "recoil"
import Participantes from "."

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
})

describe('a pagina de configuracao',()=>{
    test('deve ser renderizada corretamente',()=>{
        const { container } = render(
            <RecoilRoot>
                <Participantes/>
            </RecoilRoot>)
            expect(container).toMatchSnapshot()        
    })
})