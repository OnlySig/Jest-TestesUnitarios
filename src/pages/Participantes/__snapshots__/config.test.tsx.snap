// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <header
    class="sc-beySPh bsjzSc"
  >
    <img
      alt="logo do siteDahora"
      src="Logo Branco - Horizontal 2.svg"
    />
    <img
      alt="um cara feliz"
      class="sc-guDLey gjEtAq"
      src="carinhaDahora.svg"
    />
  </header>
  <section
    class="sc-fsYfdN bzCmZT"
  >
    <div
      class="sc-qZrbh idFbJ"
    >
      <form>
        <h1
          class="sc-dmyCSP qtfrP"
        >
          Vamos come√ßar!
        </h1>
        <div
          class="sc-jTQCzO fCYqRp"
        >
          <input
            class="inputStyled"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="sc-hLQSwg gqOfCs"
            disabled=""
          >
            Adicionar
          </button>
        </div>
        <h2
          class="sc-gLLuof iAyCrF"
        />
      </form>
      <ul
        class="sc-jsEeTM iSgqsd"
      />
      <footer
        class="sc-iBdnpw bJtMxv"
      >
        <button
          class="sc-eDLKkx cDPfUe"
          disabled=""
        >
           
          <img
            alt="iconePlay"
            src="play_circle_outline.svg"
          />
           Iniciar brincadeira!
        </button>
        <img
          alt="imagem de duas sacolas de compras"
          src="Wavy Buddies Shopping Bags.svg"
        />
      </footer>
    </div>
  </section>
</div>
`;
